exist-core/src/test/xquery/xquery3/bang.xql

```diff
@@ -202,3 +202,90 @@ declare
 function bang:element-left-number-attribute-right() {
     <emp id='1'/> ! number(@id)
 };
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:map-get () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! map:get(., 'a')
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-get () {
+    ([1], [2], [3]) ! array:get(., 1)
+};
+
+declare
+     %test:assertEquals(1,2,3)
+function bang:map-lookup-context-item () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! .?a
+};
+
+declare
+     %test:assertEquals(1,2,3)
+function bang:map-lookup-parenthesized () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! (.?a)
+};
+
+declare
+     %test:assertEquals(1,2,3)
+function bang:map-lookup-parenthesized-context-item () {
+    (map {'a':1}, map {'a':2}, map {'a':3}) ! (.)?a
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-lookup-standard () {
+    ([1], [2], [3]) ! .?1
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-lookup-parenthesized () {
+    ([1], [2], [3]) ! (.?1)
+};
+
+declare
+    %test:assertEquals(1,2,3)
+function bang:array-lookup-parenthesized-context-item () {
+    ([1], [2], [3]) ! (.)?1
+};
+
+declare
+    %test:assertEquals(1,1,1,1,1)
+function bang:mixed-function-types-call () {
+    let $id := function ($a) { $a }
+    return (function ($a) {1}, $id, sum#1, [1], map{1:1}) ! .(1)
+};
+
+declare
+    %test:assertEquals(1,1,1,1,1)
+function bang:mixed-function-types-call-parenthesized () {
+    let $id := function ($a) { $a }
+    return (function ($a) {1}, $id, sum#1, [1], map{1:1}) ! (.(1))
+};
+
+declare
+    %test:assertEquals(1,1,1,1,1)
+function bang:mixed-function-types-call-parenthesized-context-item () {
+    let $id := function ($a) { $a }
+    return (function ($a) {1}, $id, sum#1, [1], map{1:1}) ! (.)(1)
+};
+
+(:~
+ : parse error needs to be fixed first
+ : https://github.com/eXist-db/exist/issues/1655
+
+declare
+    %test:assertEquals(1,2,3)
```

Comment:
I think we discussed on the call - uncommenting these and making them as pending instead?

Reply:
No, these cannot be uncommented. Due to parser errors they entire module will not be executed.

Reply:
Ah! Okay :-)